src/mpi/topo/dims_create.c

```diff
@@ -571,9 +566,8 @@ static int optbalance(int n, int idx, int nd, int ndivs, const int divs[],
             *curbal_p = n - min;
         }
     }
-    return 0;
   fn_fail:
-    return mpi_errno;
+    MPIR_Assert(mpi_errno == MPI_SUCCESS);
```

Comment:
I'm not really liking this approach. There are only 2 callers of this function, so I think it shouldn't be too much work to do the right thing. AFAICT, the only thing that can fail is the memory allocation.

Reply:
I agree. Updated last commit. Now doing the right thing 😄